### PR TITLE
Implementation of Dockerfile

### PR DESCRIPTION
**Pull request by HumphreyHCB**
implementation of docker file. Allows a developer of the project to run the docker file to build a container with all of the necessary dependencies required to run the project, without the hassle of installing them on their local machine. 

i think additional information might be needed in the README to explain how a developer can deploy and use the docker version of the project. 

to Build the container
`docker build -t *name-the-container* .`

to Run the container with ssh
`docker run -p 33333:33333 -i -t *name-the-container*`

once in the container you will need to start the postgres server
`service postgresql start`

if all is working well and you are inside the ssh of the container, you can run
`npm run nodemon`

I strongly recommend the two VScode extensions: 
https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-docker
is useful for starting/stop containers 

https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers
allows you to open the repository of a docker container inside VScode, which not only gives you a great text editor to alter files of the project, but also allows a developer to make use of source control of the project while being inside a docker container.

atm i have yet to tests adding data to the postgres server

### CODE REVIEW

**Comment by smarr** [comment]
I did a few more changes, though, this will need a bit more work. For instance, we need to create a user with a password for the database.

File: Dockerfile [hunk 1]
```diff
@@ -4,7 +4,8 @@ FROM ubuntu:22.04
 ARG DEBIAN_FRONTEND=noninteractive
 
 # tools needed for docker setup
-RUN apt-get update && apt-get install -y apt-utils curl bash
+RUN apt-get update && apt-get install -y apt-utils curl bash && apt-get -y install sudo
+
```
Comment (smarr) [inline Dockerfile:8]:
```suggestion
RUN apt-get update && apt-get install -y apt-utils curl bash sudo
```

For simplicity, let's do it in the same `apt-get` command.

File: Dockerfile [hunk 2]
```diff
@@ -29,7 +30,7 @@ RUN npm run compile
 
 # Initialize Database
 RUN service postgresql start && \
-  PGPASSWORD=postgres psql -U postgres -c "CREATE USER docker with password 'docker'; CREATE DATABASE rebenchdb; GRANT ALL PRIVILEGES ON DATABASE rebenchdb TO docker;"
+   sudo -u postgres psql -c "CREATE USER docker with password 'docker';" -c " CREATE DATABASE rebenchdb;" -c "GRANT ALL PRIVILEGES ON DATABASE rebenchdb TO docker;"
```
Comment (smarr) [inline Dockerfile:33]: 🤢 ☹️  thanks for debugging this.

**Comment by HumphreyHCB** [comment]
i will squash all of this history one i am done and its working.
apologies the commit names might be getting a bit short

**Comment by smarr** [comment]
Think it's ready to merge.

For anyone who wants to use ReBenchDB at this point, the steps would be:

```bash
git clone https://github.com/ReBenchDB.git
cd ReBenchDB
docker build . -f ./Dockerfile -t rebenchdb
docker run -p 33333:33333 rebenchdb:latest
```

And then one can open http://localhost:33333 in the browser.

At this point, there's no much to see, but putting together a full tutorial with using ReBench is for another day, hopefully soon...

**Pull request merged**